### DISCUSSION TITLE
Why union type for members with the same name doesn't behave similar to intersection type?

### DISCUSSION THREAD

**Discussion by user753** [scala/scala3#14174] [General Discussion]
```scala
trait Foo1:
  def foo: Int

trait Foo2:
  def foo: String
    
val foo1and2: (Foo1 & Foo2) = ???
val x: Int & String = foo1and2.foo  
```
But for
```scala
val foo1or2: (Foo1 | Foo2) = ???
val x: Int | String = foo1or2.foo  
```
there is an error 
`value foo is not a member of Foo1 | Foo2`

**Comment by bishabosha** [comment]
I believe because if `foo` defined in `Foo1` and `Foo2` are not defined in the same shared parent class then the methods are actually unrelated overloads, and `foo1or2.foo` would require a reflective call

**Reply by joroKr21** [reply]
Maybe it could expand to:
```scala
foo1or2 match {
  case x: Foo1 => x.foo
  case x: Foo2 => x.foo
}
```

but that would be a feature request

**Comment by SethTisue** [comment]
https://github.com/lampepfl/dotty/discussions is for "compiler internals Q&A"

for questions, please use https://users.scala-lang.org (or see https://www.scala-lang.org/community for other options)
